shared/chat/manage-channels/edit-channel-container.js

```diff
@@ -55,10 +53,8 @@ const mergeProps = (stateProps, dispatchProps, {routeState}) => {
     channelName: stateProps.channelName,
     topic: stateProps.topic,
     onCancel: dispatchProps.onCancel,
-    onDelete: () => dispatchProps._onDelete(stateProps.teamname),
-    showDelete: false,
-    // TODO enable this after we get a better popup story
-    // showDelete: stateProps.canDelete,
+    onConfirmedDelete: () => dispatchProps.onConfirmedDelete,
+    showDelete: stateProps.canDelete,
```

Comment:
`canDelete` isn't being set here if the team details haven't been loaded yet, because we don't have role info. this causes the delete button to not show up unless they visit the teams tab > team details first. 

this is an issue that's maybe outside the scope of this ticket, but if you want to you could call `getDetails` if the team hasn't been loaded yet.

what do you think @chrisnojima? I also encountered this working with @cecileboucheron last week on something in info-panel that needed the current role. maybe we can get it from core / put in `inboxState`?

Reply:
I added a `getDetails` call and a comment. But yeah, maybe we need a more general solution...